### PR TITLE
Drive with targeting

### PR DESCRIPTION
**Pull request by hazeltorek**
includes victoria limelight code and jacobs rotate to angle code

### CODE REVIEW

File: src/main/java/frc/robot/vision/Limelight.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    public boolean isLookingLeft() {
+        if (getHorizontalOffset() < 0) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
```
Comment (hazeltorek) [inline src/main/java/frc/robot/vision/Limelight.java]: dont "if boolean return true else return false"

```java
return getHorizontalOffset() < 0;
```

File: src/main/java/frc/robot/subsystems/DriveSystem.java [hunk 2]
```diff
@@ -75,13 +186,54 @@ public void toggleSlowMode() {
...
+        mecanumDrive.driveCartesian(0, 0, 0.4);
+      }
+      else if(desiredAngle > 180)
+      {
+      mecanumDrive.driveCartesian(0, 0, -0.4);
+      }
+    }
+
+    currentAngle = gyro.getAngle();
+  }
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/DriveSystem.java]: can we change this method impl to use a pid controller like the targeting drive method?

File: src/main/java/frc/robot/subsystems/DriveSystem.java [hunk 3]
```diff
@@ -4,41 +4,88 @@
...
     mecanumDrive = new MecanumDrive(frontLeft, backLeft, frontRight, backRight);
-    gyro = new ADXRS450_Gyro();
+    gyro = new ADIS16470_IMU();
+
+    odometry = new MecanumDriveOdometry(KINEMATICS, new Rotation2d(gyro.getAngle()));
+    trajectoryConfig = new TrajectoryConfig(MAX_SPEED, MAX_ACCELERATION);
+
+    rotationController = new ProfiledPIDController(0, 0, 0, new TrapezoidProfile.Constraints(MAX_SPEED, MAX_ACCELERATION));
+
+    targetingController = new ProfiledPIDController(0, 0, 0, new TrapezoidProfile.Constraints(MAX_SPEED, MAX_ACCELERATION));
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/DriveSystem.java]: change to use rotationcontroller

File: src/main/java/frc/robot/subsystems/DriveSystem.java [hunk 4]
```diff
@@ -4,41 +4,88 @@
...
+  private ADIS16470_IMU gyro;
+
+  private MecanumDrive mecanumDrive;
+
+  private MecanumDriveOdometry odometry;
+  private TrajectoryConfig trajectoryConfig;
+
+  private ProfiledPIDController rotationController;
+
+  private ProfiledPIDController targetingController;
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/DriveSystem.java]: delete and replace usage with targetingController

File: src/main/java/frc/robot/subsystems/DriveSystem.java [hunk 5]
```diff
@@ -75,13 +186,54 @@ public void toggleSlowMode() {
...
+
+    currentAngle = gyro.getAngle();
+  }
+
+  public double getGyro() {
+    return gyro.getAngle();
+  }
+
+  public void driveWithTargeting(double x, double y, double targetAngle) {
+    drive(x / 2, y / 2, (targetingController.calculate(getGyro(), getGyro() - targetAngle)));
```
Comment (hazeltorek) [inline src/main/java/frc/robot/subsystems/DriveSystem.java]: change to use rotationController

File: src/main/java/frc/robot/subsystems/DriveSystem.java (hunk 2 above)
Comment (hazeltorek) [reply src/main/java/frc/robot/subsystems/DriveSystem.java]: nevermind

File: src/main/java/frc/robot/subsystems/DriveSystem.java [hunk 6]
```diff
@@ -75,13 +182,54 @@ public void toggleSlowMode() {
...
+      mecanumDrive.driveCartesian(0, 0, -0.4);
+      }
+    }
+
+    currentAngle = gyro.getAngle();
+  }
+
+  public double getGyro() {
+    return gyro.getAngle();
+  }
```
Comment (20MatthewNeal) [inline src/main/java/frc/robot/subsystems/DriveSystem.java]: Comment return type (radians/degrees)

**Pull request closed without merging**